### ISSUE TITLE
Add a /report in the lobby

### ISSUE THREAD

**Issue by Blocksnmore** [Minehut/Meta#415]
### Feature request

**Checklist**
- [true] I have read and agree to the README (found on https://github.com/Minehut/Meta)
- [!false] This is not a bug report, support request or plugin request
- [this != null] I have searched the Issue Tracker to make sure this is not a duplicate issue
- [() == ()] This issue does not contain any sensitive information 

**Platform**

Lobby servers (Java)
**Describe the feature**

Add a /report command on the lobbies to report players
**How the feature is useful**

When staff aren't on chat is flooded with `/join server` or `discord.gg/server` so there should be a /report command to let staff know when they join to check on said players. This would help prevent constant spam in chat since it would bring staffs attention
---
**Use 👍 and 👎 reactions to indicate you agree or disagree with this suggestion! Feel free to discuss the idea in the comments.**

**Comment by Sniped** [comment]
Where exactly would staff receive these alerts? What makes this any different from notifying a moderator on discord?

**Comment by Blocksnmore** [comment]
> Where exactly would staff receive these alerts? What makes this any different from notifying a moderator on discord?

I was thinking it would alert staff on mh since sometimes they aren't on the lobby

**Comment by DaRealMinecraftexpert** [comment]
> Where exactly would staff receive these alerts? What makes this any different from notifying a moderator on discord?

Some MH users don't have Discord, and if they catch a Rule Breaker, and need to Report it, it'd take a bit too long to Register for Discord, Join MH Discord, and then Report.

**Comment by Blocksnmore** [comment]
The mh bot could also log the reports in a channel when they are created which would be more useful for punishing players faster

**Comment by ahoulton** [comment]
Hmm, I was thinking maybe (trying) to get the reported user's chat logs and putting it on a private paste service and then sharing the link with moderators, will need a code rewrite, to be honest, and yes, this will be only lobby chatlogs.

**Comment by LukasmanMHdude** [comment]
good suggestion, upvoted

**Comment by Sniped** [comment]
> The mh bot could also log the reports in a channel when they are created which would be more useful for punishing players faster

That's what I was thinking, however not sure how the integration would work between Minehut and the bot nor if it would even be possible.

> Hmm, I was thinking maybe (trying) to get the reported user's chat logs and putting it on a private paste service and then sharing the link with moderators, will need a code rewrite, to be honest, and yes, this will be only lobby (and maybe /msg) chatlogs.

I find this the best way to go about it to be honest, and then it could just notify online staff members at the time of what's going on, or it could be on the discord bot.

**Comment by jellz** [comment]
For the record, I suggested a /chatreport command back on Discord a while back ([image](https://i.imgur.com/sWtRRy5.png) | [message](https://discord.com/channels/546414872196415501/592130982451216402/689043112382693399)), with this text:

> I know there's been plenty of cases where people have seen a spammer in the lobby, or someone making threats, or breaking any other chat rule in the lobby, making the environment toxic for players, but they weren't bothered to go to the forum, copy the report format, gather the correct evidence, make a new thread and fill it out. They'd rather just go on their way to their server. Some people just message online moderators on the Discord with a screeenshot instead of reporting it on the forum, but this channel isn't the best solution because a) the staff member could be busy or b) it's harder to keep track of reports, which is why we use the forum for reporting.
>
>I'm proposing a command that allows players to easily report a player for breaking public chat rules - /chatreport. If a player is  breaking chat rules in the lobby and no staff are in the lobby, a player can just use /chatreport <player> and a snapshot of their last 30-50 messages(?) will be saved and it will be submitted. Then, when a staff member is in the lobby they can view the reports (GUI maybe?) and view the messages, then decide the best course of action to take with punishing the player. This would make it much easier to report players, while still being able to keep track of reports.
>
>TL;DR: alternative to reporting on forum, /chatreport to report players in-game, save old messages for staff to check later

It received 36 upvotes and 0 downvotes.

---

> That's what I was thinking, however not sure how the integration would work between Minehut and the bot nor if it would even be possible.

There are a few ways this could be done, just a matter of picking the right one that works with Minehut's existing infrastructure.

**Comment by ahoulton** [comment]
> I find this the best way to go about it to be honest, and then it could just notify online staff members at the time of what's going on, or it could be on the discord bot.

Hmm, maybe could a message in-game for online staff saying "%reportee% has been reported by %reporter%, check Discord for more information!" And keep the Staff Discord as a place where reports can be logged in case someone does not admit to their punishment and showing the full report in-game would just flood the chat.

**Comment by Sniped** [comment]
> Hmm, maybe could a message in-game for online staff saying "%reportee% has been reported by %reporter%, check Discord for more information." And keep the Staff Discord as a place where reports can be logged in case someone does not admit to their punishment and showing the full report in-game would just flood the chat.

I feel like it could just go in-game as well as discord. Instead of it flooding chat, there could be text such as `Click here for more information!` that would open a GUI with further information on the report. They would also be able to click on some sort of item in the GUI that would give them a link to their past messages.

**Comment by not-3than** [comment]
Honestly this is a good idea, it's very tedious to have to report someone on the forums if they are breaking the rules. I feel only more complicated reports should take place on the forums.
